Library/Homebrew/extend/os/mac/formula.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class Formula
+  class << self
+    def uses_from_macos(dep, **args)
+      depends_on(dep) if add_mac_dependency?(args)
```

Comment:
Agreed with @xu-cheng this should should tags like `:test` and `:build` and pass them through accordingly and it should be on `SoftwareSpec`.

Reply:
If `dep` is a hash, I think that does happen with the current code (untested).

Reply:
I think it already works with tags, i'll add some specs for it